kie-server-parent/kie-server-services/kie-server-services-common/src/main/java/org/kie/server/services/impl/KieServerLocator.java

```diff
@@ -15,19 +15,38 @@
 
 package org.kie.server.services.impl;
 
+import org.kie.server.api.KieServerConstants;
+import org.kie.server.services.impl.storage.cloud.KieServerStateCloudRepository;
+
 public class KieServerLocator {
 
-    private static final KieServerImpl INSTANCE = new KieServerImpl();
-    
+    private static final KieServerImpl INSTANCE;
+
     static {
+        String strategy = System.getProperty(KieServerConstants.KIE_SERVER_STARTUP_STRATEGY);
+        
+        if (strategy == null || strategy.length() == 0) {
+            strategy = System.getenv("KIE_SERVER_STARTUP_STRATEGY");
+
+            if (strategy == null || strategy.length() == 0) {
+                strategy = ControllerBasedStartupStrategy.class.getSimpleName();
+            }
+
+            System.setProperty(KieServerConstants.KIE_SERVER_STARTUP_STRATEGY, strategy);
+        }
+
+        if ("OpenShiftStartupStrategy".equals(strategy) || "KubernetesStartupStrategy".equals(strategy)) {
+            INSTANCE = new KieServerImpl(new KieServerStateCloudRepository());
```

Comment:
this should be moved to KieServerImpl and be based on discovery mechanism - meaning if there is particular impl found (and enabled) then it should be used instead of default instead of being explicitly created